addonscript/src/addonscript/xpi.py

```diff
@@ -36,6 +17,6 @@ def get_langpack_info(path):
         "version": manifest_info["version"],
         "id": browser_specific_settings["gecko"]["id"],
         "unsigned": path,
-        "min_version": browser_specific_settings["gecko"].get("strict_min_version", get_stripped_version(manifest_info["version"])),
+        "min_version": browser_specific_settings["gecko"]["strict_min_version"],
```

Comment:
strict_min_version is not a required key in the langpack manifest. Is there a need to add an assertion for the validity here?

The current langpack generation script does always insert the key: https://searchfox.org/mozilla-central/rev/611660bff9e6d52f1769bf216a7fbd12ece4d2d2/python/mozbuild/mozbuild/action/langpack_manifest.py#482, but an extra check may still be useful.

Reply:
Breaking things loudly if `langpack_manifest.py` suddenly drops the `strict_min_version` prop is what we want here, but that kind of changes would break other things anyway.

Reply:
I don't think asserting the presence of strict_min_version is necessary, it doesn't really matter whether we're raising a KeyError or something else there.
I don't know if we want to assert something about what the version looks like, before we send it to AMO as a new firefox version, e.g. "it looks like `{major}.0`".
@diox does AMO do any validation on what the application version string looks like (in https://addons-server.readthedocs.io/en/latest/topics/api/v4_frozen/applications.html#put--api-v4-applications-(string-application)-(string-version)- specifically)?

Reply:
In addition, it generates 2 additional special versions from the one you pass using that regexp, for internal needs:
```python
release_version = '%d.%d' % (version_data['major'], version_data['minor1'] or 0)
star_version = '%d.*' % version_data['major']
```

Reply:
Thanks, that seems like it should be good enough.